### ISSUE TITLE
Re-upload build artifacts

### ISSUE THREAD

**Issue by mre** [bjorn3/rustc_codegen_cranelift#1210]
At the moment the artifact upload is disabled in the workflow here:
https://github.com/bjorn3/rustc_codegen_cranelift/blob/5f5a2f262f1d73355c8d782ec65f3774db728a4d/.github/workflows/main.yml#L155-L162
This might be a [leftover of the Windows build](https://github.com/bjorn3/rustc_codegen_cranelift/issues/1193#issuecomment-898946974).
Any chance we can re-enable the upload and trigger a new build? This would greatly help newcomers get started with cranelift as they wouldn't have to compile it on their machines first.

**Comment by bjorn3** [comment]
Artifact upload is enabled for linux and macos. Windows support is currently subpar, but I can try enabling artifact uploads for windows.

**Comment by mre** [comment]
Thanks. On that note, would it be feasible to use Github releases for the artifacts in the future? Asking because the binaries from CI expire after a while and can then no longer be downloaded. See [here](https://github.com/bjorn3/rustc_codegen_cranelift/actions/runs/1399583259) for example, which is the last successful build from a month ago. (Artifacts don't get attached to failing builds.)

**Comment by bjorn3** [comment]
I enabled artifact uploads. I had to disable compression as xzip isn't supported by windows tar and bzip2 hangs. I hope to finish https://github.com/rust-lang/rust/pull/81746 soon. This will cause cg_clif to be distributed with rustup, so `rustup component add rustc_codegen_cranelift-preview` will be enough to install it. Somewhat ironically it is currently stuck on the windows installer failing to find a file though.
